addons/mail/static/src/core/common/res_users_model.js

```diff
@@ -1,13 +1,21 @@
-import { Record } from "@mail/core/common/record";
+import { fields, Record } from "@mail/core/common/record";
 
 export class ResUsers extends Record {
     static _name = "res.users";
     static id = "id";
 
     /** @type {number} */
     id;
+    /** @type {boolean} */
+    is_admin;
     /** @type {string} */
     name;
+    /** @type {"email" | "inbox"} */
+    notification_type;
+    /** @type {boolean} false when the user is an internal user, true otherwise */
```

Comment:
```suggestion
    /** @type {boolean|undefined} false when the user is an internal user, true otherwise */
```

Given the conditions on `=== false`, `undefined` seems to matter even if I don't think it should.

Reply:
The main use case of undefined is for when the `main_user_id` is not set, not for when `share` is not set. But both are important.

It's not specific to this fields, all fields in JS are concerned by the issue when checking for "not falseness" of a boolean field that can be not set.

Reply:
We are currently very bad at handling `undefined` in the code.
It works most of the time when the positive value "unlocks" something, and we want to do the same thing when the value is not known and when the value is falsy.

For this field, more often than not, it is the negative value that "unlocks" something, and we don't want to do the same thing when it is not known.

Reply:
I get it, but the docstring is still wrong/misleading. Sure we could always consider the `|undefined` is implicit but we should write correct types when possible.